tests/unit_tests/constraint/test_skeleton.py

```diff
@@ -0,0 +1,121 @@
+import networkx as nx
+import numpy as np
+import pandas as pd
+import pytest
+
+from dodiscover import Context
+from dodiscover.ci import GSquareCITest, Oracle
+from dodiscover.constraint.skeleton import LearnSkeleton
+from dodiscover.constraint.utils import dummy_sample
+from dodiscover.testdata.testdata import bin_data, dis_data
+
+
+def common_cause_and_collider():
+    """Test orienting a common cause and a collider.
+
+    The following graph has some complexities to test the PC algorithm
+    with the Oracle setting: ``1 <- 0 -> 2 <- 3``.
+    """
+    # build initial DAG
+    ed1, ed2 = ({}, {})
+    incoming_graph_data = {0: {1: ed1, 2: ed2}, 3: {2: ed2}}
+    G = nx.DiGraph(incoming_graph_data)
+    return G
+
+
+def collider():
+    # build initial DAG
+    edge_list = [
+        ("x", "y"),
+        ("z", "y"),
+    ]
+    G = nx.DiGraph(edge_list)
+    return G
+
+
+def complex_graph():
+    """Complex graph from Figure 2 of :footcite:`Colombo2012`.
+
+    Does not include the latent edges.
+
+    References
+    ----------
+    .. footbibliography::
+    """
+    edge_list = [
+        ("x4", "x1"),
+        ("x2", "x5"),
+        ("x3", "x2"),
+        ("x3", "x4"),
+        ("x2", "x6"),
+        ("x3", "x6"),
+        ("x4", "x6"),
+        ("x5", "x6"),
+    ]
+    G = nx.DiGraph(edge_list)
+    return G
+
+
+@pytest.mark.parametrize(
+    ("indep_test_func", "data_matrix", "g_answer"),
+    [
+        (
+            GSquareCITest(),
+            np.array(bin_data).reshape((5000, 5)),
+            nx.DiGraph(
+                {
+                    0: (1,),
+                    1: (),
+                    2: (3, 4),
+                    3: (1, 2),
+                    4: (1, 2),
+                }
+            ),
+        ),
+        (
+            GSquareCITest("discrete"),
+            np.array(dis_data).reshape((10000, 5)),
+            nx.DiGraph(
+                {
+                    0: (2,),
+                    1: (2, 3),
+                    2: (),
+                    3: (),
+                    4: (3,),
+                }
+            ),
+        ),
+    ],
+)
+def test_learn_skeleton_with_data(indep_test_func, data_matrix, g_answer):
+    """Test PC algorithm for estimating the causal DAG."""
+    # the tests get the correct independences when alpha is at 0.01
+    alpha = 0.01
```

Comment:
Hmm, 'tuning' the parameter in way to make the test work is maybe not a good idea. There might be something wrong with the implementation, which is then not visible due to the tune parameter. If this is purely due to the flakiness of the (statistical) tests, consider rerunning them them when they fail with newly generated data. If it fails, e.g., 3x in a row, then it is a good indication to double check the implementation (and this could be the independence test here). See for instance https://github.com/py-why/dowhy/blob/main/tests/gcm/independence_test/test_kernel.py#L12, where we use flaky for exactly these kind of flaky scenarios.

Reply:
IIRC I think I took the data from pcalg of the R package... so it's not actually generated on the fly.

I assumed it was fine since the tests passed other types of simulated data, but let me actually investigate. 

I might just play around w/ simulating some new data instead from an SCM.

Reply:
Oh wait I just checked again and 0.05 works. I think this might've been a different private project that I was working on. The alpha level works for 0.05 (default) and 0.01.